joblib/test/test_numpy_pickle.py

```diff
@@ -680,6 +682,120 @@ def test_compression_using_file_extension():
             os.remove(dump_fname)
 
 
+@with_numpy
+def test_file_handle_persistence():
+    objs = [np.random.random((10, 10)),
+            "some data",
+            np.matrix([0, 1, 2])]
+    fobjs = [open]
+    if not PY26:
+        fobjs += [bz2.BZ2File, gzip.GzipFile]
+    if PY3_OR_LATER:
+        import lzma
+        fobjs += [lzma.LZMAFile]
+    filename = env['filename'] + str(random.randint(0, 1000))
+
+    for obj in objs:
+        for fobj in fobjs:
+            with fobj(filename, 'wb') as f:
+                numpy_pickle.dump(obj, f)
+
+            # using the same decompressor prevents from internally
+            # decompress again.
+            with fobj(filename, 'rb') as f:
+                obj_reloaded = numpy_pickle.load(f)
+
+            # when needed, the correct decompressor should be used when
+            # passing a raw file handle.
+            with open(filename, 'rb') as f:
+                obj_reloaded_2 = numpy_pickle.load(f)
+
+            if isinstance(obj, np.ndarray):
+                np.testing.assert_array_equal(obj_reloaded, obj)
+                np.testing.assert_array_equal(obj_reloaded_2, obj)
+            else:
+                nose.tools.assert_equal(obj_reloaded, obj)
+                nose.tools.assert_equal(obj_reloaded_2, obj)
+
+            os.remove(filename)
+
+
+@with_numpy
+def test_in_memory_persistence():
```

Comment:
The `io.BytesIO` object doesn't take a filename or opening mode as parameters and the same instance cannot be used from 2 different context (the first one closes it and it cannot be reopened afterwards).